### PR TITLE
feat: add login-idp-link-email page

### PR DESCRIPTION
**Pull request by Tasyp**
Adds `login-idp-link-email.ftl` page.

However, there is one issue. In the original template, the code looks like the following:
```
${msg("emailLinkIdp1", idpAlias, brokerContext.username, realm.displayName)}
```

Is it possible to somehow access the `brokerContext` inside kecloakify?

### CODE REVIEW

File: src/lib/components/LoginIdpLinkEmail.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    return (
+        <Template
+            {...{ kcContext, ...props }}
+            doFetchDefaultThemeResources={true}
+            headerNode={msg("emailLinkIdpTitle", idpAlias)}
+            formNode={
+                <>
+                    <p id="instruction1" className="instruction">
+                        {msg("emailLinkIdp1", idpAlias, realm.displayName)}
```
Comment (garronej) [inline src/lib/components/LoginIdpLinkEmail.tsx:20]:
```suggestion
                        {msg("emailLinkIdp1", idpAlias, brokerContext.username, realm.displayName)}
```

File: src/lib/components/LoginIdpLinkEmail.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { memo } from "react";
+import { Template } from "./Template";
+import type { KcProps } from "./KcProps";
+import type { KcContextBase } from "../getKcContext/KcContextBase";
+import { useKcMessage } from "../i18n/useKcMessage";
+
+export const LoginIdpLinkEmail = memo(({ kcContext, ...props }: { kcContext: KcContextBase.LoginIdpLinkEmail } & KcProps) => {
+    const { url, realm, idpAlias } = kcContext;
```
Comment (garronej) [inline src/lib/components/LoginIdpLinkEmail.tsx:8]:
```suggestion
    const { url, realm, brokerContext, idpAlias } = kcContext;
```

File: src/lib/getKcContext/KcContextBase.ts [hunk 3]
```diff
@@ -215,6 +216,11 @@ export declare namespace KcContextBase {
         idpAlias: string;
     };
 
+    export type LoginIdpLinkEmail = Common & {
+        pageId: "login-idp-link-email.ftl";
+        idpAlias: string;
+    };
```
Comment (garronej) [inline src/lib/getKcContext/KcContextBase.ts:222]:
```suggestion
    export type LoginIdpLinkEmail = Common & {
        pageId: "login-idp-link-email.ftl";
        idpAlias: string;
        brokerContext: {
            username: string;
        };
    };
```

**Review by garronej — COMMENTED** [review]
HI @Tasyp,  
Than you for your contribution!  
Would you mind trying with the following suggestions?

File: src/lib/getKcContext/KcContextBase.ts (hunk 3 above)
Comment (garronej) [reply src/lib/getKcContext/KcContextBase.ts:222]: https://github.com/InseeFrLab/keycloakify/blob/main/src/lib/getKcContext/kcContextMocks/kcContextMocks.ts

Should be updated as well.  
You should add mock data for the new page.  🙏🏻

**Comment by Tasyp** [comment]
@[garronej](https://github.com/garronej) thanks for your quick feedback!

I have addressed your feedback. It should look good now

**Comment by garronej** [comment]
Looks great, thank you very much for your contribution, it is much appreciated.

**Comment by garronej** [comment]
a new version incorporating your contribution has been released

**Pull request merged**